### PR TITLE
Rnafusion cli options cleanup

### PR DESCRIPTION
**Pull request by fevac**
## Description

This PR changes are aimed at code reorganization and the adjustment of CLI options related to Nextflow and NF-Tower.

### Changed

- Deletion of the file cg/cli/workflow/nextflow/__init__.py.
- Renaming of cg/cli/workflow/nextflow/options.py to cg/cli/workflow/nf_analysis.py, with content updates to handle CLI options for Nextflow and NF-Tower.
- Removal of OPTION_TOWER and OPTION_STUB CLI options

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b rnafusion_cli_options_cleanup -a
    ```

### How to test

- [x] Test that options are correctly reported:

```
$cg workflow rnafusion start --help

Usage: cg workflow rnafusion start [OPTIONS] CASE_ID

  Start full workflow for CASE ID.

Options:
  --log PATH           Set nextflow log file path
  --work-dir PATH      Directory where intermediate result files are stored
  --profile TEXT       Choose a configuration profile
  --config PATH        Nextflow config file path
  --params-file PATH   Nextflow pipeline-specific parameter file path
  --revision TEXT      Revision of workflow to run (either a git branch, tag
                       or commit SHA number)
  --compute-env TEXT   Compute environment name. If not specified the primary
                       compute environment will be used.
  --use-nextflow       Execute pipeline using nextflow  [default: False]
  --genomes_base PATH  Path to references folder
  -d, --dry-run        Runs the command without making any changes
  --help               Show this message and exit.
```

```
$cg workflow rnafusion run --help

Usage: cg workflow rnafusion run [OPTIONS] CASE_ID

  Run rnafusion analysis for given CASE ID.

Options:
  --log PATH          Set nextflow log file path
  --work-dir PATH     Directory where intermediate result files are stored
  --from-start        Start pipeline from start without resuming execution
                      [default: False]
  --profile TEXT      Choose a configuration profile
  --config PATH       Nextflow config file path
  --params-file PATH  Nextflow pipeline-specific parameter file path
  --revision TEXT     Revision of workflow to run (either a git branch, tag or
                      commit SHA number)
  --compute-env TEXT  Compute environment name. If not specified the primary
                      compute environment will be used.
  --use-nextflow      Execute pipeline using nextflow  [default: False]
  --nf-tower-id TEXT  NF-Tower ID of run to relaunch. If not provided the
                      latest NF-Tower ID for a case will be used.
  -d, --dry-run       Runs the command without making any changes
  --help              Show this message and exit.
```

### Expected test outcome

- [x] Check that options are excluded

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/workflow/nf_analysis.py [hunk 1]
```diff
@@ -26,27 +21,12 @@
...
     help="Choose a configuration profile",
 )
-OPTION_TOWER = click.option(
-    "--with-tower",
-    is_flag=True,
-    default=False,
-    show_default=True,
-    help="Monitor workflow execution with Seqera Tower service",
-)
-OPTION_STUB = click.option(
```
Comment (rannick) [inline cg/cli/workflow/nf_analysis.py:36]: Won't we need this for development sometimes? Is there a way to trigger a stub run otherwise?

**Review by rannick — APPROVED** [review]

File: cg/cli/workflow/nf_analysis.py (hunk 1 above)
Comment (rannick) [reply cg/cli/workflow/nf_analysis.py:36]: I have used it in the beginning, mostly when I wanted to test file delivery. I am not sure it is relevant any more. On the pure pipeline development side, I don't think it is necessary

**Review by diitaz93 — APPROVED** [review]
Great job with this clean-up 👍 I believe that providing additional details about the files and directories that were removed in the PR description would greatly contribute to a smoother review process.

**Comment by fevac** [comment]
- Deployed stage:
image

- Deployed prod:
image

**Pull request merged**